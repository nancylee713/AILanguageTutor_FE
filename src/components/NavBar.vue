<template>
  <b-navbar>
        <template slot="brand">
            <b-navbar-item tag="router-link" :to="{ path: '/' }">
                <img
                    src='../assets/teacher.png'
                    alt="Teacher logo"
                >
            </b-navbar-item>
        </template>
        <template slot="start">
            <b-navbar-item href="#">
                <router-link to='/'>
                    Home
                </router-link>
            </b-navbar-item>
            <b-navbar-item href="#">
                <router-link to='/grammar'>
                    Grammar Tests
                </router-link>
            </b-navbar-item>
            <b-navbar-item href="#">
                <router-link to='/pronunciation'>
                    Pronunciation Tests
                </router-link>
            </b-navbar-item>
            <b-navbar-dropdown label="Account">
                <b-navbar-item href="#">
                    View User Account
                </b-navbar-item>
                <b-navbar-item href="#">
                    Log Out
                </b-navbar-item>
            </b-navbar-dropdown>
        </template>

        <template slot="end">
            <b-navbar-item tag="div">
                <div class="buttons">
                    <router-link to='/create_user'>
                        <a class="button is-primary">
                            <strong>Sign up</strong>
                        </a>
                    </router-link>
                    <a class="button is-light">
                        Log in
                    </a>
                </div>
            </b-navbar-item>
        </template>
    </b-navbar>
</template>

<script>
export default {
  name: 'NavBar',
  props: {
  },
};
</script>
