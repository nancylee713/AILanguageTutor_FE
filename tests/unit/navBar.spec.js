import { shallowMount } from '@vue/test-utils';
import NavBar from '@/components/NavBar.vue';

describe('NavBar Component', () => {
  let wrapper; 

  beforeEach(() => {
    wrapper = shallowMount(NavBar);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});