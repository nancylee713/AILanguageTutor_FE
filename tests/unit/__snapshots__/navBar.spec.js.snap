// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should match snapshot 1`] = `<b-navbar><template></template> <template></template> <template></template></b-navbar>`;
